orc8r/cloud/go/services/orchestrator/servicers/grpc_exporter_servicer.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2020 The Magma Authors.
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package servicers
+
+import (
+	"context"
+	"time"
+
+	"magma/orc8r/cloud/go/services/metricsd/protos"
+
+	"github.com/golang/glog"
+	"github.com/prometheus/client_model/go"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/backoff"
+
+	edgeHubPB "github.com/facebookincubator/prometheus-edge-hub/grpc"
+)
+
+const (
+	grpcMaxTimeoutSec = 60
+	grpcMaxDelaySec   = 20
+	grpcMaxMsgSize    = 1024 * 1024 * 1024 // 1 Gb
+)
+
+type GRPCPushExporterServicer struct {
+	PushAddresses []string
+	GrpcClient    EdgeHubClient
+}
+
+type EdgeHubClient interface {
+	Collect(ctx context.Context, in *edgeHubPB.MetricFamilies, opts ...grpc.CallOption) (*edgeHubPB.Void, error)
+}
+
+// NewGRPCPushExporterServicer returns an exporter pushing metrics to
+// prometheus-edge-hubs at the given addresses.
+func NewGRPCPushExporterServicer(pushAddrs []string) protos.MetricsExporterServer {
+	conn, err := grpc.Dial(pushAddrs[0], getDialOptions()...)
```

Comment:
shouldn't we check `pushAddrs` size not to cause a panic in case it is empty?

Reply:
Good point. I think I'll just change this to be a string instead of str array since there currently isn't a usecase for multiple push endpoints